api/app/src/routes/medical/document.ts

```diff
@@ -30,11 +31,30 @@ router.get(
     const documents = await getDocuments({ patientId });
     const documentsDTO = toDTO(documents);
 
-    const queryStatus = forceQuery
-      ? await queryDocumentsAcrossHIEs({ cxId, patientId, facilityId })
-      : (await getPatientOrFail({ cxId, id: patientId })).data.documentQueryStatus ?? "completed";
+    let query: DocumentQueryResp;
 
-    return res.status(OK).json({ queryStatus, documents: documentsDTO });
+    if (forceQuery) {
+      query = await queryDocumentsAcrossHIEs({ cxId, patientId, facilityId });
+    } else {
+      const patient = await getPatientOrFail({ cxId, id: patientId });
+
+      if (patient.data.documentQueryStatus === "processing") {
+        query = {
+          queryStatus: patient.data.documentQueryStatus,
+          queryProgress: patient.data.documentQueryProgress,
+        };
+      } else {
+        query = {
+          queryStatus: "completed",
+        };
+      }
```

Comment:
NABD, but we could reuse `command/medical/document/document-query.ts`'s `createQueryResponse()`, right?